### PR TITLE
Registry contract

### LINKED ISSUE #1023: [Basic][Evmos] Order Registry contract
Add a separate contract to store all the orders.
It's intended to be used by all the devs this way:
 - Dev deploys an Order Factory / Order
 - The Order Factory deploys the Order and registers its address in the Order Registry

It must emit the corresponding event when Order is registered.

### PR DESCRIPTION
**Pull request by mn13**
## Release Notes

- **New Features**
  - Introduced a `Registry` contract for managing execution addresses and transactions.
  - Added error handling for invalid registry addresses and transaction management.
  - Enhanced `BasicOrder` contract with a new execution return type and access to unsigned transaction data.
  - Updated the `OrderFactory` to streamline interactions with the `Registry`.

- **Bug Fixes**
  - Improved validation checks for registry and transaction handling to prevent invalid operations.

- **Tests**
  - Expanded test coverage for the new `Registry` functionalities, ensuring robust error handling and interactions.

### CODE REVIEW

**Review by backsapc — APPROVED** [review]

**Comment by Svetomech** [comment]
LGTM. Though some of the coderabbit suggestions seem reasonable, can you comment on them?

**Review by Svetomech — APPROVED** [review]
LGTM! Though some of the coderabbit suggestions seem reasonable, can you comment on them?

File: solidity/orders/src/Registry.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            if (!executed) {
+                revert NotExecuted();
+            }
+        } catch {
+            revert InvalidExecutionAddress();
+        }
+
+        transactions[txHash] = execution.getTx();
+
+        emit NewTx(msg.sender, txHash);
```
[the remark this answers is not in the document]
Comment (mn13) [reply solidity/orders/src/Registry.sol:69]: We don't want to restrict transaction length and type for registry contract.

(but it is reasonable to check zero length may be add in new PR)

File: solidity/orders/src/Registry.sol [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            if (executed) {
+                revert Executed();
+            }
+        } catch {
+            revert InvalidExecutionAddress();
+        }
+
+        executions[execution] = msg.sender;
+        emit Registered(msg.sender, execution);
+    }
```
[the remark this answers is not in the document]
Comment (mn13) [reply solidity/orders/src/Registry.sol:42]: Registry is permissionless by design

File: solidity/orders/src/Registry.sol [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                revert NotExecuted();
+            }
+        } catch {
+            revert InvalidExecutionAddress();
+        }
+
+        transactions[txHash] = execution.getTx();
+
+        emit NewTx(msg.sender, txHash);
+    }
```
[the remark this answers is not in the document]
Comment (mn13) [reply solidity/orders/src/Registry.sol:70]: As written above, we don't restrict transaction type here

**Pull request merged**